im3components/wrf_tell/wrf_to_tell_part_two.py

```diff
@@ -0,0 +1,149 @@
+SWDOWN = pd.DataFrame()
+GLW = pd.DataFrame()
+
+# Loop over all the files in the list and extract then concatenate the meteorological variables:
+for file in range(len(list_of_files)):
+    # Read in the .csv file and replace missing values with nan:
+    wrf_data = pd.read_csv(list_of_files[file]).replace(-9999,np.nan)
+
+    # Drop all the rows with data outside of the 50 states:
+    wrf_data = wrf_data[wrf_data.County_FIPS <= 56045]
+
+    # Pull out a vector of the County_FIPS variable for use in subsequent subsetting:
+    if file == 0:
+       met_variable_mapping = pd.DataFrame(wrf_data['County_FIPS'].copy())
+
+    # Extract the time string from the .csv filename:
+    match = re.search(r"((\d+)_(\d+)_(\d+)_(\d+))",list_of_files[file])
+    Time[file] = pd.DataFrame({'Date':[str(dt.strptime(match.group(1),'%Y_%m_%d_%H'))]})
+
+    # Extract the meteorological variables and concatenate them into new dataframes:
+    T2[file] = wrf_data['Mean_T2']
+    Q2[file] = wrf_data['Mean_Q2']
+    SWDOWN[file] = wrf_data['Mean_SWDOWN']
+    GLW[file] = wrf_data['Mean_GLW']
+    # Compute the total wind speed based on the U10 and V10 variables:
+    WSPD[file] = ((wrf_data['Mean_U10']**2) + (wrf_data['Mean_V10']**2))**(1/2)
+
+    # Clean up the mess and move on to the next file in the list:
+    del wrf_data,match
+
+# Make a list of the unique BAs in mapping_df:
+unique_bas = mapping_df.BA_Number.unique()
+unique_bas = unique_bas[unique_bas != 14725] # Remove PJM because it's throwing a weird matrix size error
+unique_bas = unique_bas[unique_bas != 28503] # Remove WACM because it's throwing a weird matrix size error
+unique_bas = unique_bas[unique_bas != 59504] # Remove SWPP because it's throwing a weird matrix size error
+
+# Loop over the list of unique BAs and compute the population-weighted meteorology time-series:
+for ba in range(len(unique_bas)):
+    # Initialize an empty dataframe to store the results:
+    output_df = pd.DataFrame()
+
+    # Subset the mapping_df dataframe to only the counties in the given BA:
+    mapping_df_subset = mapping_df[mapping_df['BA_Number'].isin([unique_bas[ba]])]
+
+    # Find the indices of all the counties in the WRF dataset associated with that BA:
+    matched_counties = met_variable_mapping['County_FIPS'].isin(mapping_df_subset['County_FIPS'])
+
+    # Subset each of the WRF variables to only those counties: TO-DO - Make this a function
+    T2_Subset = T2.loc[matched_counties]
+    Q2_Subset = Q2.loc[matched_counties]
+    WSPD_Subset = WSPD.loc[matched_counties]
+    SWDOWN_Subset = SWDOWN.loc[matched_counties]
+    GLW_Subset = GLW.loc[matched_counties]
+
+    # Copy the time variable to the output dataframe:
+    output_df['Time'] = Time.copy().transpose()
+
+    # Multiply the WRF variable by the population fraction for that county, sum over the rows, and round off the results:
+    output_df['Mean_T2'] = T2_Subset.mul(mapping_df_subset['Population_Fraction'].values,axis=0).sum().round(2)
```

Comment:
Need spacing after commas

Reply:
Need to log a warning that the data is being rounded from its original precision